sdk/src/udp_impl.c

```diff
@@ -116,34 +116,48 @@ int mcm_send_buffer_udp(void* conn_ctx, mcm_buffer* buf)
 {
     int ret = 0;
     udp_context* ctx = NULL;
-    void* data = NULL;
-    ssize_t n = 0, len = 0;
+    char* data = NULL; // Changed to char* for pointer arithmetic
+    ssize_t n = 0;
+    size_t len = 0; // Changed to size_t to match sendto's expected type
 
     if (conn_ctx == NULL || buf == NULL) {
         log_error("Illegal Parameter.");
         return -1;
     }
-
     ctx = (udp_context*)conn_ctx;
-
     data = buf->data;
-    len = buf->len;
-    if(len < 0) {
+    len = (size_t)buf->len; // Cast to size_t, ensure that buf->len is validated before casting
+
+    if (buf->len < 0) {
```

Comment:
Shouldn't we check len here instead of buf->len? Or opposite in line 136.

Reply:
True and now it is resolved.